python/lsst/qa/explorer/writeQATable.py

```diff
@@ -0,0 +1,108 @@
+"""Command-line task and associated config for writing QA tables.
+
+The deepCoadd_qa table is a table with QA columns of interest computed
+for all filters for which the deepCoadd_obj tables are written. 
+"""
+from lsst.daf.persistence.butler import Butler
+from lsst.pex.config import (Config, Field, ConfigField, ListField, DictField, ConfigDictField,
+                             ConfigurableField)
+from lsst.pipe.base import Task, CmdLineTask, ArgumentParser, TaskRunner, TaskError
+from lsst.coadd.utils import TractDataIdContainer
+from lsst.pipe.tasks.multiBand import MergeSourcesTask, MergeSourcesConfig
+from lsst.pipe.tasks.multiBand import _makeGetSchemaCatalogs
+from lsst.coadd.utils.coaddDataIdContainer import ExistingCoaddDataIdContainer
+
+import functools
+import re, os
+import pandas as pd
+
+from .parquetTable import ParquetTable
+from .catalog import MultilevelParquetCatalog
+from .functors import CompositeFunctor, RAColumn, DecColumn
+
+# Question: is there a way that LSST packages store data files?
+ROOT = os.path.abspath(os.path.dirname(__file__))
+
+class PostprocessCatalogTask(CmdLineTask):
+    """Base class for postprocessing calculations on catalogs
+
+    """    
+
+    inputDataset = 'deepCoadd_obj'
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        """Create a suitable ArgumentParser.
+        
+        """
+        parser = ArgumentParser(name=cls._DefaultName)
+        parser.add_id_argument("--id", cls.inputDataset,
+                               ContainerClass=ExistingCoaddDataIdContainer,
+                               help="data ID, e.g. --id tract=12345 patch=1,2")
+        return parser
+
+
+    def run(self, patchRef):
+        """Do calculations; write result
+        """
+        df = self.doCalculations(patchRef)
```

Comment:
`run` should do the `patchRef.get(self.inputDataset)` call and pass the result (presumably a `ParquetTable` object, so the I/O is still deferred) to `doCalculations`, so the `Task` can be used (via `doCalculations`) either without a butler or without any I/O (when passed a ParquetTable that already has the necessary columns in memory, as it would if it was about to be written for the first time).

Eventually, it'd be nice to make `inputDataset` into `inputDatasets`, and have it take a sequence of datasets that would be read individually by `run` and merged into a single `ParquetTable` object.  I don't think that's needed immediately, though, and we may want to spec out the namespace/structure for both datasets and DataFrame a bit more fully before thinking about merging.

Reply:
I originally had what you suggested, and changed it to this because I need to access `.dataId['patch']` to add the `patchId` column to the dataframes.  Perhaps I can pass both the result of `.get()` and a `dataId` to `run`?  That's probably ok.